File: app/src/main/java/com/example/leafy/model/PlantNetResponse.kt
package com.example.leafy.model

data class PlantNetResponse(
    val results: List<PlantNetResult> = emptyList()
)

// PlantNet suele devolver score como número (double)
data class PlantNetResult(
    val score: Double = 0.0,
    val species: Species = Species()
)

/**
 * OJO: aquí añadimos family y genus como objetos anidados.
 * Si por alguna razón no vienen en la respuesta, quedan en null.
 */
data class Species(
    val scientificName: String = "",
    val commonNames: List<String> = emptyList(),
    val family: Taxon? = null,
    val genus: Taxon? = null
)

data class Taxon(
    val scientificName: String = ""
)

File: lib/models/plant_net_response.dart
class PlantNetResponse {
  final List<PlantNetResult> results;

  const PlantNetResponse({this.results = const []});

  factory PlantNetResponse.fromJson(Map<String, dynamic> json) {
    final list = (json['results'] as List<dynamic>? ?? [])
        .map((e) => PlantNetResult.fromJson(e as Map<String, dynamic>))
        .toList();
    return PlantNetResponse(results: list);
  }
}

class PlantNetResult {
  final double score;
  final Species species;

  const PlantNetResult({this.score = 0.0, this.species = const Species()});

  factory PlantNetResult.fromJson(Map<String, dynamic> json) {
    return PlantNetResult(
      score: (json['score'] as num?)?.toDouble() ?? 0.0,
      species: json['species'] != null
          ? Species.fromJson(json['species'] as Map<String, dynamic>)
          : const Species(),
    );
  }
}

class Species {
  final String scientificName;
  final List<String> commonNames;
  final Taxon? family;
  final Taxon? genus;

  const Species({
    this.scientificName = '',
    this.commonNames = const [],
    this.family,
    this.genus,
  });

  factory Species.fromJson(Map<String, dynamic> json) {
    return Species(
      scientificName: json['scientificName']?.toString() ?? '',
      commonNames: (json['commonNames'] as List<dynamic>? ?? [])
          .map((e) => e.toString())
          .toList(),
      family: json['family'] != null
          ? Taxon.fromJson(json['family'] as Map<String, dynamic>)
          : null,
      genus: json['genus'] != null
          ? Taxon.fromJson(json['genus'] as Map<String, dynamic>)
          : null,
    );
  }
}

class Taxon {
  final String scientificName;

  const Taxon({this.scientificName = ''});

  factory Taxon.fromJson(Map<String, dynamic> json) {
    return Taxon(scientificName: json['scientificName']?.toString() ?? '');
  }
}
